tests/integration/gcs_asset_test.py

```diff
@@ -0,0 +1,258 @@
+import time
+from typing import Generator
+
+import pytest
+
+from pyatlan.client.atlan import AtlanClient
+from pyatlan.errors import AtlanError, ErrorCode, NotFoundError
+from pyatlan.model.assets import Asset, Connection, GCSBucket, GCSObject
+from pyatlan.model.core import Announcement
+from pyatlan.model.enums import (
+    AnnouncementType,
+    AtlanConnectorType,
+    CertificateStatus,
+    EntityStatus,
+)
+from pyatlan.model.response import AssetMutationResponse
+from tests.integration.client import TestId, delete_asset
+from tests.integration.connection_test import create_connection
+
+MODULE_NAME = TestId.make_unique("GCS")
+
+CONNECTOR_TYPE = AtlanConnectorType.GCS
+GCS_BUCKET_NAME = MODULE_NAME
+GCS_OBJECT_NAME = f"{MODULE_NAME}.csv"
+CERTIFICATE_STATUS = CertificateStatus.VERIFIED
+CERTIFICATE_MESSAGE = "Automated testing of the Python SDK."
+ANNOUNCEMENT_TYPE = AnnouncementType.INFORMATION
+ANNOUNCEMENT_TITLE = "Python SDK testing."
+ANNOUNCEMENT_MESSAGE = "Automated testing of the Python SDK."
+
+
+def block(
+    client: AtlanClient, response: AssetMutationResponse
+) -> AssetMutationResponse:
+    if response.mutated_entities and response.mutated_entities.DELETE:
+        _retrieve_and_check(client, response.mutated_entities.DELETE, 0)
+    return response
+
+
+def _retrieve_and_check(client: AtlanClient, to_check: list[Asset], retry_count: int):
+    leftovers = []
+    for one in to_check:
+        try:
+            candidate = client.asset.get_by_guid(one.guid, asset_type=type(one))
+            if candidate and candidate.status == EntityStatus.ACTIVE:
+                leftovers.append(candidate)
+        except NotFoundError:
+            # If it is not found, it was successfully deleted (purged), so we
+            # do not need to look for it any further
+            print("Asset no longer exists.")
+        except AtlanError:
+            leftovers.append(one)
+    if leftovers:
+        if retry_count == 20:
+            raise ErrorCode.RETRY_OVERRUN.exception_with_parameters()
+        time.sleep(2)
+        _retrieve_and_check(client, leftovers, retry_count + 1)
```

Comment:
This code is repeated in multiple integration tests. Please refactor it into tests/integretaion/utils.py.